test/parallel/test-http-pipeline-regr-3332.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+const net = require('net');
+
+const big = new Buffer(16 * 1024);
+big.fill('A');
+
+const COUNT = 1e4;
+
+var received = 0;
+
+var client;
+const server = http.createServer(function(req, res) {
+  res.end(big, function() {
+    if (++received === COUNT) {
+      server.close();
+      client.end();
+    }
+  });
+}).listen(common.PORT, function() {
+  var req = new Array(COUNT + 1).join('GET / HTTP/1.1\r\n\r\n');
+  client = net.connect(common.PORT, function() {
+    client.write(req);
+  });
```

Comment:
Won't having no `'data'` event cause all the data to buffer? Don't think that's what you're trying to test, and that would accumulate to ~150MB. On our machines that's nothing big, but I'm thinking about execution time. It's not huge, but is longer than the rest:
```
ok 56 test-http-pipeline-regr-3332.js
  ---
  duration_ms: 42.108
```